pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcSendingMailbox.java

```diff
@@ -131,9 +132,8 @@ public void cancel(Throwable t) {
     try {
       String msg = t != null ? t.getMessage() : "Unknown";
       // NOTE: DO NOT use onError() because it will terminate the stream, and receiver might not get the callback
-      _contentObserver.onNext(toMailboxContent(
-          ErrorMseBlock.fromException(new RuntimeException("Cancelled by sender with exception: " + msg, t)),
-          List.of()));
+      _contentObserver.onNext(toMailboxContent(ErrorMseBlock.fromException(
+          new QueryCancelledException("Cancelled by sender with exception: " + msg)), List.of()));
```

Comment:
Nit: you can just create the error map here or create a new constructor factory method that accepts the error code and message instead of allocating a new exception that will be not referenced after that.

Reply:
Yeah, well seen. Replaced with a `ErrorMseBlock.fromError(code, msg)` method.